shtab/__init__.py

```diff
@@ -474,24 +474,29 @@ def is_opt_end(opt):
     def is_opt_multiline(opt):
         return isinstance(opt, OPTION_MULTI)
 
-    def format_optional(opt):
+    def format_optional(opt, parser):
+        formatter = parser._get_formatter()
+        formatter._width = 1000000
```

Comment:
You are right, I think it is only used in `_format_text(desc)` but I was just being safe, maybe we should factor this in a `get_formatter(parser)` function, also formatter can be customized by projects, so it could be used in a user defined formatter.

https://github.com/iterative/shtab/blob/f9de186008d3394609206e115d7c120443e364dd/shtab/__init__.py#L551